nessie/src/test/java/org/apache/iceberg/nessie/TestNessieIcebergClient.java

```diff
@@ -91,6 +111,412 @@ public void testWithReferenceAfterRecreatingBranch()
+                  .containsEntry(CatalogProperties.APP_ID, "iceberg-nessie");
+            },
+            Index.atIndex(1));
+  }
+
+  @Test
+  public void testDropNamespaceNotEmpty() throws NessieConflictException, NessieNotFoundException {
+    String branch = "dropNamespaceInvalidBranch";
+    createBranch(branch);
+    NessieIcebergClient client = new NessieIcebergClient(api, branch, null, Map.of());
+
+    client.createNamespace(Namespace.of("a"), Map.of());
+    client.createNamespace(Namespace.of("a", "b"), Map.of());
+
+    Assertions.assertThatThrownBy(() -> client.dropNamespace(Namespace.of("a")))
+        .hasMessageContaining("Namespace 'a' is not empty.");
+  }
+
+  @Test
+  public void testDropNamespaceConflict() throws NessieConflictException, NessieNotFoundException {
+    String branch = "dropNamespaceConflictBranch";
+    createBranch(branch);
+    NessieIcebergClient client = new NessieIcebergClient(api, branch, null, Map.of());
+
+    client.createNamespace(Namespace.of("a"), Map.of());
+
+    client.commitTable(
+        null, newTable(), "file:///tmp/iceberg", (String) null, ContentKey.of("a", "tbl"));
+
+    Assertions.assertThatThrownBy(() -> client.dropNamespace(Namespace.of("a", "tbl")))
+        .hasMessageContaining("Content object with name 'a.tbl' is not a Namespace.");
+  }
+
+  @Test
+  public void testDropNamespaceExternalConflict()
+      throws NessieConflictException, NessieNotFoundException {
+    String branch = "dropNamespaceExternalConflictBranch";
+    createBranch(branch);
+    NessieIcebergClient client = new NessieIcebergClient(api, branch, null, Map.of());
+
+    client.createNamespace(Namespace.of("a"), Map.of());
+
+    org.projectnessie.model.Namespace original = fetchNamespace(ContentKey.of("a"), branch);
+    org.projectnessie.model.Namespace updated =
+        org.projectnessie.model.Namespace.builder()
+            .from(original)
+            .properties(Map.of("k1", "v1"))
+            .build();
+    commit(branch, "update namespace a", Operation.Put.of(ContentKey.of("a"), updated));
+
+    Assertions.assertThatThrownBy(() -> client.dropNamespace(Namespace.of("a")))
+        .hasMessageContaining(
+            "Cannot drop Namespace 'a': Values of existing and expected content for key 'a' are different.");
+  }
+
+  @Test
+  public void testDropNamespaceNonExistingRef()
+      throws NessieConflictException, NessieNotFoundException {
+    String branch = "dropNamespaceNonExistingRefBranch";
+    createBranch(branch);
+    NessieIcebergClient client = new NessieIcebergClient(api, branch, null, Map.of());
+
+    client.createNamespace(Namespace.of("a"), Map.of());
+
+    api.deleteBranch().branch((Branch) api.getReference().refName(branch).get()).delete();
+
+    Assertions.assertThat(client.dropNamespace(Namespace.of("a"))).isFalse();
+  }
+
+  @Test
+  void testSetProperties() throws NessieConflictException, NessieNotFoundException {
```

Comment:
any particular reason why all other test methods are public and these are not? I would align them to be public